nvtabular/ops.py

```diff
@@ -1242,30 +1239,32 @@ def op_logic(self, gdf: cudf.DataFrame, target_columns: list, stats_context={}):
                 new_gdf[new_col] = new_gdf[new_col].astype(self.dtype, copy=False)
         return new_gdf
 
-    def get_emb_sz(self, encoders, cat_names):
-        # sorted key required to ensure same sort occurs for all values
-        ret_list = [
-            (n, self.def_emb_sz(encoders, n))
-            for n in sorted(cat_names, key=lambda entry: entry.split("_")[0])
-        ]
-        return ret_list
 
-    def emb_sz_rule(self, n_cat: int) -> int:
-        return min(16, round(1.6 * n_cat ** 0.56))
+def get_embedding_order(cat_names):
+    """ Returns a consistent sorder order for categorical variables
 
-    def def_emb_sz(self, classes, n, sz_dict=None):
-        """Pick an embedding size for `n` depending on `classes` if not given in `sz_dict`.
-        """
-        sz_dict = sz_dict if sz_dict else {}
-        n_cat = classes[n]
-        sz = sz_dict.get(n, int(self.emb_sz_rule(n_cat)))  # rule of thumb
-        self.embed_sz[n] = sz
-        return n_cat, sz
-
-    def get_embeddings(self, encoders, cat_names):
-        embeddings = {}
-        for col in sorted(cat_names):
-            path = encoders[col]
-            num_rows, _, _ = cudf.io.read_parquet_metadata(path)
-            embeddings[col] = (num_rows, self.emb_sz_rule(num_rows))
-        return embeddings
+    Parameters
+    -----------
+    cat_names : list of str
+        names of the categorical columns
+    """
+    return sorted(cat_names, key=lambda entry: entry.split("_")[0])
```

Comment:
Where is this strange sort key coming from? If this requirement is coming from some under-the-hood library, then are we relying on stable sorting to break ties? E.g. imagine column names `'user_feature_1'` and `'user_feature_2'`.

Reply:
@benfred not sure if you're notified without this mention

Reply:
These are good points, the delimiter chosen is bad and the sort is not the best. This was created while specifically working on the criteo outbrains and rossman datasets. For those targets, the delimiter there was just in place to separate the operators from the columns when you do not replace are target column in an operator. This trickles down. I agree that this should be changed. But maybe not in this PR. Because currently that is how non-replaced columns are created. Essentially columns are named based on where in the chain of operations they are created,  appending operators chained to represent a given columns data. i.e. col_op1_op2_op3_... So if we change this we need to change to handled that as well. Otherwise we will break operator(replace=False) functionality.

Reply:
If it breaks other functionality I definitely agree – let's leave it out of this PR.

However can you please elaborate on what would break? I'm pretty new to NVTabular so I don't see why a typical sort would break anything (using the entire string as the key).

E.g. imagine columns `['user_feature_1', 'user_feature_2', 'user_feature_3']` and applying an operator with `replace=True` to form `'user_feature_2'`. Then we'd end up with a sorted list `['user_feature_1', 'user_feature_2_op1', 'user_feature_3']`. What would break in this case?

Reply:
No thats not what would happen. the moment you set `replace=True` the output will be `['user_feature_1', 'user_feature_2', 'user_feature_3']` esesntially just the original columns... if you did `replace=False` after operator you would have `['user_feature_1', 'user_feature_2', 'user_feature_3', 'user_feature_2_op1']`. But to your question above, the reason for the sort is to guarantee an order. When I first started working with nvtabular, I was hitting an edge case (i dont remember specifically, I think it was with outbrains) where the loading of embeddings to the model and the tensors did not match up because of a difference in keyword ordering... So the sort, in every place you see it is to ensure that given a list of values, I can guarantee the order of that list. If we want to move to a different method for guaranteeing, in all cases, this functionality, I dont have a problem with that.

Reply:
Ah yes sorry, I did mean `replace=False` there (not `replace=True`)

I see your point, and I agree that a sort can be helpful to guarantee a certain global order, but my question is instead about the key used to sort: `key=lambda entry: entry.split("_")[0]`

The question I'm asking is: what would break if we were to instead a *typical* key (`entry`, e.g. `'user_feature_2'`) rather than this modified key (`entry.split("_")[0]`, e.g. `'user'` instead of `'user_feature_2'`) ?

(Also, just to note, this may seem harmless, but this led to confusion + bugs for me while exploring the Rossmann notebook)

Reply:
I think @rdipietro has a good point here about subsorting - I think it should be fine to use ```entry``` as a sort key instead of ```entry.split("_")[0]``` in all cases (because we still be sorting by the first token up to '_', and just using the rest of the key to subsort after that). @jperez999 do you have any objections to this?

Reply:
I've made that change in the most recent commit